routes/access_tokens.go

```diff
@@ -64,13 +64,17 @@ type createAccessTokenRequest struct {
 // through the same channel (looking it up by the state).
 // Create will then receive the result through the channel, remove the channel
 // from the map, and serialise a result back to the client.
-func (a AccessTokens) Create(w http.ResponseWriter, r *http.Request) {
+func (a AccessTokens) Create(w http.ResponseWriter, r *http.Request) error {
 	var req createAccessTokenRequest
 
+	logger, err := GetLogger(r)
+	if err != nil {
+		return err
+	}
+
 	if err := jsonapi.UnmarshalPayload(r.Body, &req); err != nil {
-		a.Logger.With("error", err.Error()).Info("failed to unmarshal request")
 		RenderError(w, http.StatusBadRequest, invalidJSONError)
-		return
+		return errors.Wrap(err, "failed to unmarshal request")
```

Comment:
But I think we probably shouldn't have this? I think this will attempt to render twice.

Reply:
Yep, you’re right. I think I just made a mistake here.